prov/rxd/src/rxd.h

```diff
@@ -214,58 +215,66 @@ DECLARE_FREESTACK(struct rxd_x_entry, rxd_rx_fs);
 #define rxd_ep_rx_flags(rxd_ep) ((rxd_ep)->util_ep.rx_op_flags)
 #define rxd_ep_tx_flags(rxd_ep) ((rxd_ep)->util_ep.tx_op_flags)
 
-struct rxd_ctrl_hdr {
+struct rxd_pkt_hdr {
 	uint32_t version;
+	uint32_t flags;
+	uint32_t tx_id;
+	uint32_t rx_id;
+	uint32_t key;
+	uint32_t seg_no;
+	fi_addr_t peer;
+};
+
+struct rxd_ctrl_hdr {
 	uint16_t type;
 	uint16_t window;
 	uint64_t size;
```

Comment:
This has a 32-bit gap between window and size fields.

Reply:
Ahh shoot, forgot to fix it.